pkg/agent/util/net_windows_test.go

```diff
@@ -0,0 +1,1342 @@
+			netNatName: "test-nat",
+			subnetCIDR: &net.IPNet{
+				IP:   net.ParseIP("192.168.1.21"),
+				Mask: net.CIDRMask(32, 32),
+			},
+			commandOut: "0.0.0.0/32",
+			commandErr: nil,
+			wantErr:    nil,
+		},
+		{
+			name:       "Net Nat Not Found",
+			netNatName: "test-nat",
+			subnetCIDR: &net.IPNet{
+				IP:   net.ParseIP("192.168.1.21"),
+				Mask: net.CIDRMask(32, 32),
+			},
+			commandOut: "",
+			commandErr: fmt.Errorf("invalid"),
+			wantErr:    fmt.Errorf("invalid"),
+		},
+		{
+			name:       "Net Nat Exist",
+			netNatName: "test-nat",
+			subnetCIDR: &net.IPNet{
+				IP:   net.ParseIP("192.168.1.21"),
+				Mask: net.CIDRMask(32, 32),
+			},
+			commandOut: "192.168.1.21/32",
+			commandErr: nil,
+			wantErr:    nil,
+		},
+		{
+			name:       "Net Nat Add Fail",
+			netNatName: "test-nat",
+			subnetCIDR: &net.IPNet{
+				IP:   net.ParseIP("192.168.1.21"),
+				Mask: net.CIDRMask(32, 32),
+			},
+			commandOut: "",
+			commandErr: fmt.Errorf("received error No MSFT_NetNat objects found"),
+			wantErr:    fmt.Errorf("received error No MSFT_NetNat objects found"),
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			runCommand = func(cmd string) (string, error) {
+				return tc.commandOut, tc.commandErr
+			}
+			defer func() {
+				runCommand = ps.RunCommand
+			}()
+			gotErr := NewNetNat(tc.netNatName, tc.subnetCIDR)
+			assert.Equal(t, tc.wantErr, gotErr)
+		})
+	}
+}
+
+func TestReplaceNetNatStaticMapping(t *testing.T) {
+	tests := []struct {
+		name           string
+		netNatName     string
+		externalIPAddr string
+		externalPort   uint16
+		internalIPAddr string
+		internalPort   uint16
+		proto          string
+		commandOut     string
+		commandErr     error
+		wantErr        error
+	}{
+		{
+			name:           "Replace Net Nat",
+			netNatName:     "test-nat",
+			externalIPAddr: "a.b.c.0/24",
+			externalPort:   80,
+			internalIPAddr: "192.0.02.179",
+			internalPort:   8080,
+			proto:          "tcp",
+			commandOut:     "0;1 nil nil nil 192.168.1.21 80",
+			commandErr:     nil,
+			wantErr:        nil,
+		},
+		{
+			name:           "Get Net Nat Err",
+			netNatName:     "test-nat",
+			externalIPAddr: "a.b.c.0/24",
+			externalPort:   80,
+			internalIPAddr: "192.0.02.179",
+			internalPort:   8080,
+			proto:          "tcp",
+			commandOut:     "",
+			commandErr:     fmt.Errorf("invalid"),
+			wantErr:        fmt.Errorf("invalid"),
+		},
+		{
+			name:           "Remove Net Nat Err",
+			netNatName:     "test-nat",
+			externalIPAddr: "a.b.c.0/24",
+			externalPort:   80,
+			internalIPAddr: "192.0.02.179",
+			internalPort:   8080,
+			proto:          "tcp",
+			commandOut:     "0;1 nil nil nil 192.168.1.21 80",
+			commandErr:     fmt.Errorf("received error No MSFT_NetNatStaticMapping objects found"),
+			wantErr:        fmt.Errorf("received error No MSFT_NetNatStaticMapping objects found"),
+		},
+		{
+			name:           "Add Net Nat Err",
+			netNatName:     "test-nat",
```

Comment:
Ditto, no need to define this `netNatName` if they are always the same and doesn't impact the output. Please check other variables.

Reply:
Checked.